cli/app/commands/conflict/conflict.py

```diff
@@ -0,0 +1,391 @@
+        if not requirement:
+            return requirement
+
+        # Handle npm-style caret ranges: ^1.20.0 -> >=1.20.0, <2.0.0
+        if requirement.startswith("^"):
+            base_version = requirement[1:]
+            try:
+                parsed = version.parse(base_version)
+                major = parsed.major
+                return f">={base_version}, <{major + 1}.0.0"
+            except Exception:
+                return f">={base_version}"
+
+        # Handle npm-style tilde ranges: ~1.20.0 -> >=1.20.0, <1.21.0
+        if requirement.startswith("~"):
+            base_version = requirement[1:]
+            try:
+                parsed = version.parse(base_version)
+                major = parsed.major
+                minor = parsed.minor
+                return f">={base_version}, <{major}.{minor + 1}.0"
+            except Exception:
+                return f">={base_version}"
+
+        # Handle ranges with commas: 1.20.0,2.0.0 -> >=1.20.0, <2.0.0
+        if "," in requirement and not any(op in requirement for op in [">=", "<=", ">", "<", "==", "!="]):
+            parts = [p.strip() for p in requirement.split(",")]
+            if len(parts) == 2:
+                return f">={parts[0]}, <{parts[1]}"
+
+        # Handle x.x.x format ranges: 1.20.* -> >=1.20.0, <1.21.0
+        if requirement.endswith(".*") or requirement.endswith(".x"):
+            base_version = requirement.replace(".*", ".0").replace(".x", ".0")
+            try:
+                parsed = version.parse(base_version)
+                major = parsed.major
+                minor = parsed.minor
+                return f">={base_version}, <{major}.{minor + 1}.0"
+            except Exception:
+                return f">={base_version}"
+
+        # Return as-is if it already contains operators
+        return requirement
+
+
+class ConfigLoader:
+    """Handles loading and parsing of configuration files."""
+    
+    def __init__(self, logger: LoggerProtocol):
+        self.logger = logger
+
+    def load_config(self, config_path: str) -> Dict[str, Any]:
+        """Load configuration from YAML file."""
+        self.logger.debug(conflict_loading_config.format(path=config_path))
+
+        if not os.path.exists(config_path):
+            raise FileNotFoundError(conflict_config_not_found.format(path=config_path))
+
+        try:
+            with open(config_path, "r") as f:
+                config = yaml.safe_load(f)
+
+            self.logger.debug(conflict_config_loaded)
+            return config
+        except yaml.YAMLError as e:
+            raise ValueError(conflict_invalid_config.format(error=str(e)))
+        except Exception as e:
+            raise Exception(conflict_invalid_config.format(error=str(e)))
+
+
+class ToolVersionChecker:
+    """Handles version checking for different tools."""
+    
+    # Common version commands for different tools
+    VERSION_COMMANDS = {
```

Comment:
let's have this in the yaml only, what is the intention behind keeping it here,

Reply:
Umm, for me, this seemed more of business logic, so i kept it in the code rather than in config.prod.yaml;

Considered it in pov that configuration for data, code for logic, so left it in code as it made more sense to me that way.

I can see from pov where you are coming from. i can make changes for this. but before that, what if we move out of code, and keep it as constant.py rather than config?